test/e2e/clusternetworkpolicy_test.go

```diff
@@ -0,0 +1,558 @@
+	networkPolicyDelay = 2 * time.Second
+)
+
+func failOnError(err error, t *testing.T) {
+	if err != nil {
+		log.Errorf("%+v", err)
+		k8sUtils.Cleanup(namespaces)
+		t.Fatalf("test failed: %v", err)
+	}
+}
+
+// TestCase is a collection of TestSteps to be tested against.
+type TestCase struct {
+	Name  string
+	Steps []*TestStep
+}
+
+// TestStep is a single unit of testing spec. It includes the CNP specs that need to be
+// applied for this test, the port to test traffic on and the expected Reachability matrix.
+type TestStep struct {
+	Name         string
+	Reachability *Reachability
+	CNPs         []*secv1alpha1.ClusterNetworkPolicy
+	Port         int
+	Duration     time.Duration
+}
+
+func initialize(t *testing.T, data *TestData) {
+	p80 = 80
+	p81 = 81
+	pods = []string{"a", "b", "c"}
+	namespaces = []string{"x", "y", "z"}
+
+	for _, podName := range pods {
+		for _, ns := range namespaces {
+			allPods = append(allPods, NewPod(ns, podName))
+		}
+	}
+	err := enableCNP(data)
+	failOnError(err, t)
+	k8sUtils, err = NewKubernetesUtils(data)
+	failOnError(err, t)
+	ips, err := k8sUtils.Bootstrap(namespaces, pods)
+	failOnError(err, t)
+	podIPs = *ips
+}
+
+func enableCNP(data *TestData) error {
+	configMap, err := data.GetAntreaConfigMap(antreaNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get ConfigMap: %v", err)
+	}
+	antreaControllerConf, _ := configMap.Data["antrea-controller.conf"]
+	antreaAgentConf, _ := configMap.Data["antrea-agent.conf"]
+	antreaControllerConf = strings.Replace(antreaControllerConf, "#featureGates:", "featureGates:\n   ClusterNetworkPolicy: true", 1)
+	antreaAgentConf = strings.Replace(antreaAgentConf, "#featureGates:", "featureGates:\n   ClusterNetworkPolicy: true", 1)
+	configMap.Data["antrea-controller.conf"] = antreaControllerConf
+	configMap.Data["antrea-agent.conf"] = antreaAgentConf
+	if _, err := data.clientset.CoreV1().ConfigMaps(antreaNamespace).Update(context.TODO(), configMap, metav1.UpdateOptions{}); err != nil {
+		return fmt.Errorf("failed to update ConfigMap %s: %v", configMap.Name, err)
+	}
+	_, err = data.restartAntreaControllerPod(defaultTimeout)
+	if err != nil {
+		return fmt.Errorf("error when restarting antrea-controller Pod: %v", err)
+	}
+	// TODO: restart agent pods as well if featureGate affects agent
+	return nil
+}
+
+func applyDefaultDenyToAllNamespaces(k8s *KubernetesUtils, namespaces []string) error {
+	if err := k8s.CleanNetworkPolicies(namespaces); err != nil {
+		return err
+	}
+	for _, ns := range namespaces {
+		builder := &NetworkPolicySpecBuilder{}
+		builder = builder.SetName(ns, "default-deny-namespace")
+		builder.SetTypeIngress()
+		if _, err := k8s.CreateOrUpdateNetworkPolicy(ns, builder.Get()); err != nil {
+			return err
+		}
+	}
+	time.Sleep(networkPolicyDelay)
+	r := NewReachability(allPods, false)
+	k8s.Validate(allPods, r, p80)
+	_, wrong, _ := r.Summary()
+	if wrong != 0 {
+		return fmt.Errorf("error when creating default deny k8s network policies")
```

Comment:
nit: s/network policies/NetworkPolicies for consistency

(elsewhere as well)

Reply:
Done.